### PR TITLE
client: specify the quota type when finding the root quota realm

### PR DESCRIPTION
**Pull request by lxbsz**
Currently if only the quota.files is enabled for the statfs it will
use the wrong quota realm when finding the root quota realm. And vice
versa for the rename. This will specify the quota type for different
cases.

This is from Luis' kernel client's fix.

Fixes: https://tracker.ceph.com/issues/56414
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/include/cephfs/types.h [hunk 1]
```diff
@@ -238,8 +244,16 @@ struct quota_info_t
   bool is_valid() const {
     return max_bytes >=0 && max_files >=0;
   }
-  bool is_enable() const {
-    return max_bytes || max_files;
+  bool is_enable(quota_max_t type=QUOTA_ANY) const {
+    switch (type) {
```
Comment (mchangir) [inline src/include/cephfs/types.h:248]: nit: typo in function name; rename to `is_enabled()`

**Review by kotreshhr — APPROVED** [review]
Looks good to me.

**Comment by rptaylor** [comment]
This should fix an issue affecting me; is it ready to merge?  Thanks @lxbsz

**Comment by lxbsz** [comment]
> This should fix an issue affecting me; is it ready to merge? Thanks @lxbsz

It needs the qa test.

**Review by rishabh-d-dave — APPROVED** [review]
QA was successful - https://tracker.ceph.com/projects/cephfs/wiki/Main#2022-Aug-26.

**Pull request merged**